src/agent/base_agent.py

```diff
@@ -168,15 +171,15 @@ def query(self, query: str) -> str:
         response = self.routing_agent.chat(query)
         self.routing_agent.reset()
         # write response to file for KG writeback
-        with open("data/external_response.txt", "w") as f:
+        with open("/brain/src/streamlit_app/data/external_response.txt", "w") as f:
```

Comment:
This absolute path is not acceptable for our repository.
How about setting `data_dir` argument for docker build?